core/states/navigating.py

```diff
@@ -89,7 +92,12 @@ async def run(self) -> RoverState:
             return self.on_event(core.AutonomyEvents.OBSTACLE_AVOIDANCE)
 
         # Check if we are still approaching the goal
-        if algorithms.gps_navigate.get_approach_status(goal, current, start) != core.ApproachState.APPROACHING:
+        if (
+            algorithms.gps_navigate.get_approach_status(
+                goal, current, start, 0.75 if (leg_type == "POSITION") else core.WAYPOINT_DISTANCE_THRESHOLD
```

Comment:
First, I think it was determined 0.75m is too close to the waypoint. Second, this still seems hacky. At the very least, 0.75 should be defined as a constant, but it would probably be better to have a variable threshold that is defined elsewhere

Reply:
Moving this to a separate issue